aio_pika/pool.py

```diff
@@ -0,0 +1,66 @@
+import asyncio
+import logging
+from typing import AsyncContextManager, TypeVar, Coroutine, Callable, Any
+
+
+T = TypeVar("T")
+ItemType = Coroutine[Any, None, T]
+ConstructorType = Callable[[], ItemType]
+
+
+log = logging.getLogger(__name__)
+
+
+class Pool:
+    __slots__ = (
+        'loop', '__max_size', '__items',
+        '__constructor', '__created', '__lock'
+    )
+
+    def __init__(self, constructor: ConstructorType, max_size: int = None,
+                 loop: asyncio.AbstractEventLoop = None):
+
+        self.loop = loop or asyncio.get_event_loop()
+        self.__max_size = max_size or -1
```

Comment:
It seems that `-1` is not supported by the code below.
If `self.__max_size == -1`, `self._is_overflow` will be always `True`, and `await self.__items.get()` will hang in the `_get`.